frontend/server/src/Controllers/User.php

```diff
@@ -3116,19 +3116,41 @@ public static function getRankDetailsForSmarty(\OmegaUp\Request $r) {
+     * @return array{smartyProperties: array{payload: array{coderOfTheMonthData: array{birth_date: int|null, classname: string, country: null|string, country_id: int|null, email: null|string, gender: null|string, graduation_date: int|null, gravatar_92: string, hide_problem_tags: bool|null, is_private: bool, locale: string, name: null|string, preferred_language: null|string, scholar_degree: null|string, school: null|string, school_id: int|null, state: null|string, state_id: int|null, username: null|string, verified: bool}|null, currentUserInfo: array{username?: string}, enableSocialMediaResources: bool, rankTable: array{rank: list<array{classname: string, country_id: null|string, name: null|string, problems_solved: int, rank: int, score: float, user_id: int, username: string}>, total: int}, runsChartPayload: array{date: list<string>, total: list<int>}, schoolOfTheMonthData: array{country_id: null|string, name: string, school_id: int}|null, schoolRank: list<array{country_id: string, name: string, school_id: int, score: float}>, upcomingContests: array{number_of_results: int, results: list<array{alias: string, title: string}>}}}, template: string}
      */
     public static function getIndexDetailsForSmarty(\OmegaUp\Request $r) {
         try {
             $r->ensureIdentity();
-            $isLogged = true;
         } catch (\OmegaUp\Exceptions\UnauthorizedException $e) {
             // Not logged, but there is no problem with this
-            $isLogged = false;
+            /** @var null $r->identity */
         }
         $date = !empty($r['date']) ? strval($r['date']) : null;
         $firstDay = self::getCurrentMonthFirstDay($date);
         $rowCount = 5;
+
+        $activeContests = \OmegaUp\DAO\Enum\ActiveStatus::ACTIVE;
+        $recommended = \OmegaUp\DAO\Enum\RecommendedStatus::ALL;
+        $page = 1;
+        $pageSize = 20;
+        $cacheKey = "{$activeContests}-{$recommended}-{$page}-{$pageSize}";
+        $contests = \OmegaUp\Controllers\Contest::getContestList(
+            $r->identity,
+            /*$query=*/ null,
+            $cacheKey,
+            $page,
+            $pageSize,
+            $activeContests,
+            $recommended
```

Comment:
nit: puedes poner el comentario mágico en vez de declarar las variables?

```suggestion
            /*$page=*/1,
            $pageSize,  // etc.
            $activeContests,
            $recommended
```

Reply:
Listo